### PR TITLE
Switch to custom properties to control grid gutter widths

### PR DESCRIPTION
**Pull request by MartijnCuppens**
Not sure yet how we should continue with the grid, so I'm just gonna make some improvements step by step which are easier to review/merge (unlike https://github.com/twbs/bootstrap/pull/28517 😛). In this PR I've switched to CSS custom properties which makes it easier to control the gutter width and saves a lot of CSS.

I've also moved the loop to save some media query prints.

CSS diff see: https://www.diffchecker.com/XRJUJMzm

### CODE REVIEW

**Comment by ffoodd** [comment]
LGTM.

Since we're using more and more custom properties, we'll be using more and more `calc()` function: shouldn't we allow them in stylelint from now on?

Unrelated thing, there's a flexbug fix  [L68](https://github.com/twbs/bootstrap/blob/ecfe1cb5ee4e97b792602b156e9323113d541b9a/scss/mixins/_grid.scss#L68) which only targets IE10-11. I'll do a PR latter.

**Comment by MartijnCuppens** [comment]
> Since we're using more and more custom properties, we'll be using more and more calc() function: shouldn't we allow them in stylelint from now on?

The reason we disallow them is because `calc(... + 0)` is invalid. Might be worth reviewing later on if we bump into this more than once.

**Review by ffoodd — APPROVED** [review]

File: scss/mixins/_grid.scss [hunk 1]
```diff
@@ -3,10 +3,13 @@
 // Generate semantic grid columns with these mixins.
 
 @mixin make-row($gutter: $grid-gutter-width) {
+  --grid-gutter-x: #{$gutter};
+  --grid-gutter-y: 0;
```
Comment (ffoodd) [inline scss/mixins/_grid.scss]: As a sidenote, you could remove this declaration and use `var(--grid-gutter-y, 0)` in `margin-top` to shorten things a bit. Could be useful in some cases to reduce custom properties bloat.

Not sure if it's readable enough, though.

Comment (MartijnCuppens) [reply scss/mixins/_grid.scss]: We do need to reset this custom property, otherwise styles will leak in to nested rows.

Comment (ffoodd) [reply scss/mixins/_grid.scss]: OW nested things. I need to accomodate :D

**Pull request merged**